controller/service.go

```diff
@@ -121,6 +125,26 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 				lbIPs = []net.IP{}
 			}
 		}
+		// annotationLoadBalancerIPs takes higher precedence over svc.Spec.LoadBalancerIP
+		desiredLbIPs, err := c.getDesiredLbIPs(svc)
+		if err != nil {
+			level.Error(l).Log("event", "loadbalancerIP", "error", err, "msg", "invalid requested loadbalancer IPs")
+			c.client.Errorf(svc, "LoadBalancerFailed", "invalid requested loadbalancer IPs: %s", err)
+			return true
+		}
+		if len(desiredLbIPs) > 0 {
+			ok, err := c.isLbIPsEqual(lbIPs, desiredLbIPs)
```

Comment:
can u use reflect.DeepEqual() instead of this new custom API ?

Reply:
at this point both lbIPs are unsorted, so couldn't use reflect.DeepEqual.